### ISSUE TITLE
Importing molten.openapi raise a RecursionError exception

### ISSUE THREAD

**Issue by nicolas-leydet** [Bogdanp/molten#6]
## Environment 
 * SUSE Linux Enterprise Desktop 12 SP3
 * Python 3.6.0
 * molten 0.2.1

#### bug.py
``` 
import molten.openapi
```

$ python  bug.py

## What did you expect would happen?

Do nothing but with no error

## What happened?

```
Traceback (most recent call last):
  File "simple_app.py", line 1, in <module>
    import molten.openapi
  File "~/.virtualenvs/molten-test/lib/python3.6/site-packages/molten/openapi/__init__.py", line 18, in <module>
    from .documents import (
  File "~/.virtualenvs/molten-test/lib/python3.6/site-packages/molten/openapi/documents.py", line 78, in <module>
    @schema
  File "~/.virtualenvs/molten-test/lib/python3.6/site-packages/molten/validation/schema.py", line 83, in schema
    field.select_validator()
  File "~/.virtualenvs/molten-test/lib/python3.6/site-packages/molten/validation/field.py", line 143, in select_validator
    self.validator = _select_validator(self)
  File "~/.virtualenvs/molten-test/lib/python3.6/site-packages/molten/validation/field.py", line 466, in _select_validator
    if validator.can_validate_field(field):
  File "~/.virtualenvs/molten-test/lib/python3.6/site-packages/molten/validation/field.py", line 288, in can_validate_field
    return get_origin(annotation) in (list, List)
  File "/usr/local/lib/python3.6/typing.py", line 760, in __eq__
    return self._subs_tree() == other
  File "/usr/local/lib/python3.6/typing.py", line 760, in __eq__
    return self._subs_tree() == other
  File "/usr/local/lib/python3.6/typing.py", line 760, in __eq__
    return self._subs_tree() == other
  [Previous line repeated 240 more times]
  File "/usr/local/lib/python3.6/typing.py", line 759, in __eq__
    if not isinstance(other, _Union):
RecursionError: maximum recursion depth exceeded while calling a Python object
```

**Comment by Bogdanp** [comment]
Thanks for reporting this! Unfortunately, I don't think there's anything that I can do about it since it appears to be a bug in Python 3.6.0. The issues seems to be fixed in 3.6.1 and up.

**Comment by Bogdanp** [comment]
I'm going to close this for now. Let me know if you still run into the issue on later versions of Python.
